src/theia/sfm/feature_extractor_and_matcher.cc

```diff
@@ -224,8 +282,18 @@ void FeatureExtractorAndMatcher::ProcessImage(
   // Extract Features.
   std::vector<Keypoint> keypoints;
   std::vector<Eigen::VectorXf> descriptors;
-  ExtractFeatures(options_, image_filepath, &keypoints, &descriptors);
-
+  if (mask_filepaths_.empty()) {
```

Comment:
Using the std::unordered_map as suggested above, you can use theia's theia/util/map_util.h functions:
const std::string mask_filepath = FindWithDefault(image_masks_, image_filepath, "");